app/src/org/commcare/activities/HomeScreenBaseActivity.java

```diff
@@ -201,12 +201,17 @@ private void loadInstanceState(Bundle savedInstanceState) {
     private void processFromExternalLaunch(Bundle savedInstanceState) {
         if (savedInstanceState == null && getIntent().hasExtra(DispatchActivity.WAS_EXTERNAL)) {
             wasExternal = true;
-            processSessionEndpoint();
-            sessionNavigator.startNextSessionStep();
+            if (processSessionEndpoint()) {
+                sessionNavigator.startNextSessionStep();
+            }
```

Comment:
I see that you're returning false if either `endpoint is null` or there were some `exceptions`.  Do we need to show some error dialog when the `endpoint is null`?

Reply:
No, An external launch doesn't need to be a session endpoint launch. One can simply [encode the session as a string](https://github.com/dimagi/commcare-android/blob/master/app/src/org/commcare/activities/DispatchActivity.java#L340) instead of specifying the endpoint.

Reply:
Got it, Thanks